spec/prog/postgres/converge_postgres_resource_spec.rb

```diff
@@ -16,37 +17,41 @@
     )
   }
 
-  let(:postgres_resource) {
+  let(:pg) {
     pr = PostgresResource.create(
-      name: "pg-test", superuser_password: "dummy-password", ha_type: "sync",
+      name: "pg-test", superuser_password: "dummy-password", ha_type: "none",
       target_version: "17", location_id: location_id, project_id: project.id,
       user_config: {}, pgbouncer_user_config: {}, target_vm_size: "standard-2",
       target_storage_size_gib: 64, private_subnet_id: private_subnet.id
     )
     Strand.create_with_id(pr, prog: "Postgres::PostgresResourceNexus", label: "wait")
+    Firewall.create(name: "#{pr.ubid}-internal-firewall", location_id: location_id, project_id: postgres_service_project.id)
     pr
   }
 
   let(:strand) {
     Strand.create(
       prog: "Postgres::ConvergePostgresResource", label: "start",
-      parent_id: postgres_resource.strand.id,
-      stack: [{"subject_id" => postgres_resource.id}]
+      parent_id: pg.strand.id,
+      stack: [{"subject_id" => pg.id}]
     )
   }
 
   before do
-    allow(nx).to receive(:postgres_resource).and_return(postgres_resource)
+    allow(nx).to receive(:postgres_resource).and_return(pg)
```

Comment:
Not sure whether this line is needed. If you need to mock methods on `pg`, mock them on `nx.postgres_resource`, which should be cached.